### PR TITLE
Allow specifying a tool to push with

### PR DESCRIPTION
**Pull request by kevinboulain**
Hi,

macOS includes `rsync` by default but probably not the expected version so using `push` results in:
```
push -host root@host periph.io/x/periph/cmd/periph-info
- Building periph.io/x/periph/cmd/periph-info
- Pushing 1 executables to . in root@192.168.1.107 via rsync
rsync: --info=progress2: unknown option
rsync error: syntax or usage error (code 1) at /BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-52.200.1/rsync/main.c(1337) [client=2.6.9]
push: exit status 1

Visit https://github.com/periph/bootstrap#troubleshooting-push for help.
```

I would like an overridable detection as it may help in some cases (another example may be if the remote host doesn't even have `rsync`).

Feel free to nitpick/close/rewrite/... :)

**EDIT**: and I forgot to read the issues, see #5

### CODE REVIEW

File: cmd/push/main.go [hunk 1]
```diff
@@ -177,6 +177,7 @@ func mainImpl() error {
 	tags := flag.String("tags", "", "build tags to pass")
 	rel := flag.String("rel", ".", "directory on remote host to push files into")
 	host := flag.String("host", os.Getenv("PUSH_HOST"), "host to push to; defaults to content of environment variable PUSH_HOST")
+	preferredTool := flag.String("tool", "", "tool to push with")
```
Comment (maruel) [inline cmd/push/main.go]: It's opaque for users which values are valid. Please improve the help string.

**Comment by maruel** [comment]
I installed rsync via homebrew so it works for me, but I agree it's better for the tool to just work without having to fiddle. It'd be great if it could detect the rsync version and not pass recent flags to it.

File: cmd/push/main.go (hunk 1 above)
Comment (kevinboulain) [reply cmd/push/main.go]: Done.

**Comment by kevinboulain** [comment]
> It'd be great if it could detect the rsync version and not pass recent flags to it.

I agree, but besides the fact it could be brittle, it would still not cover the case where the remote host doesn't have rsync (I use a lightweight distribution and I didn't install it by default).
That can still be easily added afterwards though.

**Comment by maruel** [comment]
Thanks!

**Pull request merged**